proto/api/v1/remote_runner.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package api.v1;
+
+import "google/rpc/status.proto";
+
+message RunRequest {
+  // URL of the repo the remote workspace should be initialized for
+  // Ex. "https://github.com/some-user/acme"
+  string repo = 1;
+
+  // Git refs to configure the remote git workspace (at least one of `branch` or
+  // `commit_sha` must be set). If only `branch` is set, will run from the tip
+  // of the branch. If only `commit_sha` is set, reporting will not contain the
+  // branch name.
+  string branch = 2;
+  string commit_sha = 3;
+
+  // Bash commands to run, in order.
+  // Ex. [{"run": "bazel build :server"}, {"run": "echo \"Hello World\""}]
+  repeated Step steps = 4;
+
+  // Environment variables to set in the runner env.
+  map<string, string> env = 5;
+
+  // Platform properties to apply to the runner.
+  // Ex. ["OSFamily":"linux"}, {"Arch":"amd64"}]
+  map<string, string> platform_properties = 6;
+
+  // Max time before run should be canceled.
+  // Ex. "15s", "2h"
+  string timeout = 7;
+
+  // If true, start the runner but do not wait for the action to be scheduled
+  // before returning a response.
+  bool async = 8;
+
+  // Whether to use github credentials configured on the executor.
+  //
+  // By default, we require https for git operations and generate short-lived
+  // access tokens using our github app installation.
+  // If GitHub SSH access is already configured on the runners, set this
+  // to true to skip doing that and use the configured auth.
+  //
+  // This is only supported for bare runners.
+  bool use_system_git_credentials = 9;
+}
+
+message Step {
+  string run = 1;
```

Comment:
Two slight preferences for run: 1. It can be a bash block and doesn't have to be a single command 2. 'run' is what github actions uses 

I don't feel strongly though. What do you think?

Reply:
Going to merge as is for now - but let me know if you feel strongly about changing it